box-browse-sdk/src/main/java/com/box/androidsdk/browse/fragments/BoxFilterSearchResultsFragment.java

```diff
@@ -0,0 +1,302 @@
+package com.box.androidsdk.browse.fragments;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.support.v4.app.Fragment;
+import android.support.v7.widget.AppCompatCheckBox;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.Button;
+import android.widget.CompoundButton;
+import android.widget.ImageView;
+import android.widget.LinearLayout;
+import android.widget.RelativeLayout;
+import android.widget.ScrollView;
+import android.widget.TextView;
+
+import com.box.androidsdk.browse.R;
+import com.box.androidsdk.browse.models.BoxSearchFilters;
+
+import java.util.HashMap;
+
+public class BoxFilterSearchResultsFragment extends Fragment {
+
+    private static String EXTRA_FILTERS = "extraFilters";
+    private BoxSearchFilters mFilters;
+    private HashMap<BoxSearchFilters.ItemType, FileTypeData> mFileTypeMap;
+    private boolean mDateModifiedExpanded;
+    private boolean mItemSizeExpanded;
+    private ScrollView mScrollView;
+    private LinearLayout mDateModifiedView;
+    private LinearLayout mSizeView;
+
+    public BoxFilterSearchResultsFragment() {
+        // Required empty public constructor
+    }
+
+
+    public static BoxFilterSearchResultsFragment newInstance(BoxSearchFilters filters) {
+        BoxFilterSearchResultsFragment fragment = new BoxFilterSearchResultsFragment();
+        Bundle args = new Bundle();
+        args.putSerializable(EXTRA_FILTERS, filters);
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (getArguments() != null) {
+            mFilters = (BoxSearchFilters) getArguments().getSerializable(EXTRA_FILTERS);
+        }
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        // Inflate the layout for this fragment
+        View view = inflater.inflate(R.layout.fragment_box_filter_search_results, container, false);
+        mScrollView = (ScrollView) view.findViewById(R.id.scrollView);
+        mDateModifiedView = (LinearLayout) view.findViewById(R.id.dateModified);
+        mSizeView = (LinearLayout) view.findViewById(R.id.size);
+
+        mDateModifiedExpanded = false;
+        mItemSizeExpanded = false;
+        setup(view);
+
+        return view;
+    }
+
+    public BoxSearchFilters getCurrentFilters() {
+        return mFilters;
+    }
+
+    private void setup(View view) {
+        setupButtons(view);
+        setupFileTypes(view);
+        setupDateModified(view);
+        setupSizeRange(view);
+    }
+
+    private void setupSizeRange(View view) {
+        setupSizeRange(view, BoxSearchFilters.ItemSize.Any, R.id.itemSizeContainerAny, R.string.item_size_any);
```

Comment:
This could be more concise if you had a collection of size range. Think I had this comment when this code was still part of the app, but a lot of the collections in this class seem to fit a list/adapter pattern; so the views can be decoupled from the data.